pkg/apis/rollouts/v1alpha1/types.go

```diff
@@ -144,6 +144,48 @@ type CanaryStrategy struct {
 	MaxSurge *intstr.IntOrString `json:"maxSurge,omitempty"`
 }
 
+// RolloutCanaryExperimentStep defines a template that is used to create a experiment for a step
+type RolloutCanaryExperimentStep struct {
```

Comment:
I can imagine a future where ExperimentStep is not just being done as part of Canary, but BlueGreen. May want to name this just RolloutExperimentStep to not box ourselves.

Reply:
Done